### PR TITLE
Update with the changes in sobel gradients

### LINKED ISSUE #5500: Dim error in `HoVerNetLoss`
**Describe the bug**
Since in `SobelGradients `, we need input with shape (C, H, W, [D]), but I find in `HoVerNetLoss`, it used shape with (B, H, W).
https://github.com/Project-MONAI/MONAI/blob/27362e482761985daac8d5ef732eb57b50c741f1/monai/transforms/post/array.py#L888-L890
https://github.com/Project-MONAI/MONAI/blob/27362e482761985daac8d5ef732eb57b50c741f1/monai/apps/pathology/losses/hovernet_loss.py#L69-L72

### PR DESCRIPTION
**Pull request by drbeh**
Fixes #5500 

### Description

Update `HoVerNetLoss` with the improvements and changes in `SobelGradients`.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).

### CODE REVIEW

**Comment by drbeh** [comment]
> One question - is there any practical reason not to use the channel for the batch (apart from the fact that it is called 'channel' and not 'batch')? If not then perhaps we should generalize Sobel so that it can take either (or both). Just a thought.

That's a good point! I avoid it because of clarity but now that you are mentioning it, I think why not! I will make the change and also update the SobelGradients docstring to make it clear that it can be channel or batch!

**Review by JHancox — APPROVED** [review]

**Comment by wyli** [comment]
@KumoLiu please help check with the training pipeline

**Comment by KumoLiu** [comment]
> @KumoLiu please help check with the training pipeline

Yes, I'm checking right now!

File: monai/apps/pathology/losses/hovernet_loss.py [hunk 1]
```diff
@@ -61,15 +61,21 @@ def __init__(
 
         self.dice = DiceLoss(softmax=True, smooth_dr=1e-03, smooth_nr=1e-03, reduction="sum", batch=True)
         self.ce = CrossEntropyLoss(reduction="mean")
-        self.sobel = SobelGradients(kernel_size=5)
+        self.sobel_v = SobelGradients(kernel_size=5, spatial_axes=0)
```
[the remark this answers is not in the document]
Comment (KumoLiu) [reply monai/apps/pathology/losses/hovernet_loss.py:64]: Hi @drbeh, thanks for your explanation, just want to double-check then we will generate the gradient map in the horizontal direction, right?
243671668096719_ pic

[the remark this answers is not in the document]
Comment (drbeh) [reply monai/apps/pathology/losses/hovernet_loss.py:64]: Hi @KumoLiu, in the loss function, we calculate the horizontal gradient of the horizontal map and vertical gradient of the vertical map, right? Is there anything missing here?

[the remark this answers is not in the document]
Comment (drbeh) [reply monai/apps/pathology/losses/hovernet_loss.py:64]: @KumoLiu thank you very much for catching this bug! I fixed it.

**Comment by Nic-Ma** [comment]
> Hi @Nic-Ma, could you please help to merge this PR? Thanks

Hi @drbeh ,

https://github.com/Project-MONAI/MONAI/pull/5480 is running Blossom CI, I will merge this PR after it.

Thanks.
